### PR TITLE
Support for multiple drop message binary transport formats

### LINKED ISSUE #61: Implement a QblTransportMessage class
This class should be used between the Drop and the Crypto component. It allows the Crypto component to be agnostic about the Drop message format.

### LINKED ISSUE #251: Implement drop message padding
Implement padding of drop messages to fixed size as decided in Qabel/qabel-doc#81.

### PR DESCRIPTION
**Pull request by cburkert**
This implements
* a basic binary drop message handling API
* handling of binary drop message in the current version (0)

EDIT:
Implements message padding and closes #251 
Also implements a transport message handling thus closes #61 

Depends on Qabel/qabel-drop#14 to update the maximum drop message size on the drop server.

### CODE REVIEW

**Comment by cburkert** [comment]
After merging this PR, the travis will be happy again (as you see) and the probeTooBigMessage test passes again.

File: src/main/java/de/qabel/core/crypto/CryptoUtils.java [hunk 1]
```diff
@@ -472,11 +474,14 @@ private boolean validateSignature(byte[] message, byte[] signature,
...
 	 * @throws InvalidKeyException
 	 */
+	@Deprecated
 	public byte[] encryptHybridAndSign(String message,
 			QblEncPublicKey recipient, QblSignKeyPair signatureKey)
 			throws InvalidKeyException {
 		ByteArrayOutputStream bs = new ByteArrayOutputStream();
 		SecretKey aesKey = keyGenerator.generateKey();
+		
+		// pad message
```
Comment (k-c13) [inline src/main/java/de/qabel/core/crypto/CryptoUtils.java]: Is there something missing?

Comment (cburkert) [inline src/main/java/de/qabel/core/crypto/CryptoUtils.java]: No, this method is no longer used.

File: src/main/java/de/qabel/core/crypto/AbstractBinaryDropMessage.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+			return null;
+		}
+		// spoofing detection currently not supported
+		// if (dropMessage.getSender() != new
+		// String(sender.getSignPublicKeys().get(0).getPublicKeyFingerprint()))
+		// {
+		// // invalid sender claim
+		// logger.info("Drop message contained spoofed sender field.");
+		// return null;
+		// }
```
Comment (k-c13) [inline src/main/java/de/qabel/core/crypto/AbstractBinaryDropMessage.java]: Can this code removed?

**Comment by k-c13** [comment]
Looks good

But can you extract the DropHTTPTest from this pull request to get travis to work again

File: src/main/java/de/qabel/core/crypto/AbstractBinaryDropMessage.java (hunk 2 above)
Comment (cburkert) [inline src/main/java/de/qabel/core/crypto/AbstractBinaryDropMessage.java]: I don't think so. I plan to make this spoofing detection feature work soon.

**Comment by cburkert** [comment]
Ok, I cherry-picked it into a hotfix PR.

File: src/main/java/de/qabel/core/crypto/CryptoUtils.java [hunk 3]
```diff
@@ -44,16 +44,16 @@
 	private final static String SIGNATURE_ALGORITHM = "RSASSA-PSS";
 	private final static String RSA_CIPHER_ALGORITM = "RSA/ECB/OAEPWITHSHA1ANDMGF1PADDING";
 	private final static String HMAC_ALGORITHM = "HMac/" + "SHA512";
-	private final static int RSA_SIGNATURE_SIZE_BYTE = 256;
-	private final static String SYMM_KEY_ALGORITHM = "AES";
+	final static int RSA_SIGNATURE_SIZE_BYTE = 256;
```
Comment (L-Henke) [inline src/main/java/de/qabel/core/crypto/CryptoUtils.java]: I dont' like that ```BinaryDropMessageV0``` accesses a final static value from ```CryptoUtils```. That is unusual and not very clean. I think that shared final static values either has to be extracted into a dedicated shared configuration class or should only be access via a getter.

Comment (cburkert) [inline src/main/java/de/qabel/core/crypto/CryptoUtils.java]: Ok, I only changed the visibility because I don't also want to do CryptoUtils refactoring in this PR. I'd welcome if we could tackle this issue in another PR.

File: src/test/java/de/qabel/core/drop/DropControllerTest.java [hunk 4]
```diff
@@ -146,44 +147,6 @@ public void sendTestSingle() throws InvalidKeyException, MalformedURLException,
...
-        Assert.assertEquals(headerAndMessage[0], (byte) 1);
-        Assert.assertArrayEquals(messageBytes, messageBytesRemovedHeader);
-
-        Assert.assertEquals(dm.getTime(), newMessage.getTime());
-        Assert.assertEquals(dm.getSender(), newMessage.getSender());
-        Assert.assertEquals(dm.getAcknowledgeID(), newMessage.getAcknowledgeID());
-        Assert.assertEquals(dm.getVersion(), newMessage.getVersion());
-        Assert.assertEquals(dm.getModelObject(), newMessage.getModelObject());
-    }
-
```
Comment (thechauffeur) [inline src/test/java/de/qabel/core/drop/DropControllerTest.java]: Why did you remove this test without adding a similar test somewhere else? Also you do not test your code in this PR in any way.

File: src/main/java/de/qabel/core/crypto/CryptoUtils.java (hunk 3 above)
Comment (thechauffeur) [inline src/main/java/de/qabel/core/crypto/CryptoUtils.java]: OK.

**Comment by thechauffeur** [comment]
Besides my comments IMO this PR can be merged.

File: src/test/java/de/qabel/core/drop/DropControllerTest.java (hunk 4 above)
Comment (cburkert) [inline src/test/java/de/qabel/core/drop/DropControllerTest.java]: This scenario is pretty well covered by the DropControllerTests that already exists, as is the whole functionality of my PR. So, I don't see anything new to test, but if anybody spots a test-worthy feature, I happily make one.

**Comment by jan-schreib** [comment]
Fine with me. :+1:
